### ISSUE TITLE
Why content = enc_topk_memory.detach() that memory need to be detach() ？

### ISSUE THREAD

**Issue by LuletterSoul** [Peterande/D-FINE#121]
Thank you Open Source for such a great job. Why does the memory need to be detached? In this case, the gradient of the transformer decoder cannot be propagated to the backbone.

**Comment by SebastianJanampa** [comment]
You can find the information in this [paper](https://arxiv.org/pdf/2203.03605), section 3.5
